docs/architecture/adr-044-protobuf-updates-guidelines.md

```diff
@@ -47,10 +48,36 @@ For this reason, module developers MUST NOT add new fields to existing `Msg`s.
 
 It is worth mentioning that this does not limit adding fields to a `Msg`, but also to all nested structs and `Any`s inside a `Msg`.
 
-#### 2. Non-`Msg`-related Protobuf definitions MAY have new fields
+#### 2. Non-`Msg`-related Protobuf definitions MAY have new fields, but MUST add a `Since:` comment
 
 On the other hand, module developers MAY add new fields to Protobuf definitions related to the `Query` service or to objects which are saved in the store. This recommendation follows the Protobuf specification, but is added in this document for clarity.
 
+The SDK requires the Protobuf comment of the new field to contain one line with the following format:
+
+```protobuf
+// Since: cosmos-sdk <version>{, <version>...}
+```
+
+Where each `version` denotes a minor ("0.45") or patch ("0.44.5") version from which the field is available. This will greatly help client libraries, who can optionally use reflection or custom code generation to show/hide these fields depending on the targetted node version.
+
+As examples, the following comments are valid:
+
+```protobuf
+// Since: cosmos-sdk 0.44
+
+// Since: cosmos-sdk 0.42.11, 0.44.5
```

Comment:
Are these intended to be machine readable? If so, should the since be lower case, similar to build?

Reply:
Right now it is not read by machines but only developers who have the information available in doc comments, markdown dumps of proto docs and ideally IDEs as plain text. Long term it would be great to make this parsable.

To me upper case looks nice and how you should start a new paragraph in doc coments. However, I don't know to which "build" you are referring to. Could you link an example?